wagtail/core/tests/test_page_model.py

```diff
@@ -291,6 +294,19 @@ def test_urls_with_multiple_sites(self):
         self.assertEqual(christmas_page.relative_url(events_site), '/christmas/')
         self.assertEqual(christmas_page.get_site(), events_site)
 
+        request = HttpRequest()
+        request.site = second_events_site
+        self.assertEqual(
+            christmas_page.get_url_parts(request=request),
+            (second_events_site.id, 'http://second_events.example.com', '/christmas/')
+        )
+
+        request.site = events_site
+        self.assertEqual(
+            christmas_page.get_url_parts(),
```

Comment:
You forgot to pass the request ;)
Also, please move this test of the first event site before the test of the second event site.